tensorflow/compiler/xla/service/pattern_matcher_test.cc

```diff
@@ -1102,5 +1102,66 @@ TEST_F(PatternMatcherTest, CustomCallMatchers) {
       root, m::CustomCall("test_target", m::Parameter(1), m::Parameter(0))));
 }
 
+TEST_F(PatternMatcherTest, UnaryOpAnyOf) {
+  constexpr char kModuleStr[] = R"(
+    HloModule test_module
+
+    ENTRY test {
+      p0 = f32[] parameter(0)
+      cos = cosine(p0)
+      ROOT out = f32[] abs(cos)
+    }
+  )";
+  TF_ASSERT_OK_AND_ASSIGN(auto hlo_module,
+                          ParseAndReturnVerifiedModule(kModuleStr));
+  auto* root = hlo_module->entry_computation()->root_instruction();
+
+  EXPECT_TRUE(Match(
+      root, m::UnaryOpAnyOf(m::Abs(), {HloOpcode::kBitcast, HloOpcode::kCos})));
+  EXPECT_TRUE(
+      Match(root, m::UnaryOpAnyOf(m::Abs(m::Cos()),
+                                  {HloOpcode::kBitcast, HloOpcode::kCos})));
+  EXPECT_TRUE(Match(
+      root, m::UnaryOpAnyOf(m::Cos(), {HloOpcode::kCos, HloOpcode::kAbs})));
+  EXPECT_FALSE(Match(
+      root, m::UnaryOpAnyOf(m::Bitcast(), {HloOpcode::kCos, HloOpcode::kAbs})));
+  EXPECT_FALSE(Match(
+      root, m::UnaryOpAnyOf(m::Cos(), {HloOpcode::kCos, HloOpcode::kBitcast})));
```

Comment:
I don't think that `UnaryOpAnyOf` captures what I need in order to read this.  To me, this name sounds like "pick one of kCos, kAbs, etc" but does not capture the fact that this op is *optional* and it's ok if it's not present at all.

I hear you saying that `OptionalUnaryOp` is vague.  Can you elaborate how a reader might misinterpret a specific example like `m::OptionalUnaryOp({kCos, kBitcast}, m::Abs())`?

Reply:
It may not always be obvious what the optionality refers to, especially when the operand is also a unary op and when the list contains one element. The least ambiguous name is probably the original `UnaryOpAnyOfOrOperand`.